src/status_im2/contexts/chat/messages/composer/view.cljs

```diff
@@ -0,0 +1,250 @@
+  [{:keys [chat-id] :as params}]
+  (fn [evt]
+    (let [new-height (oget evt "nativeEvent.contentSize.height")]
+      (swap! input/input-text-content-heights assoc chat-id new-height)
+      (if @initial-content-height
+        (do
+          ;;on Android when last symbol removed, height value is wrong, like 300px
+          (when (string/blank? (get @input/input-texts chat-id))
+            (swap! input/input-text-content-heights assoc chat-id @initial-content-height))
+          (update-y params))
+        (reset! initial-content-height new-height)))))
+
+(defn mentions
+  [{:keys [refs suggestions max-y]} insets]
+  [:f>
+   (fn []
+     (let [translate-y (reanimated/use-shared-value 0)]
+       (rn/use-effect
+        (fn []
+          (reanimated/set-shared-value translate-y
+                                       (reanimated/with-timing (if (seq suggestions) 0 200)))))
+       [reanimated/view
+        {:style (reanimated/apply-animations-to-style
+                 {:transform [{:translateY translate-y}]}
+                 {:bottom     (or (:bottom insets) 0)
+                  :position   :absolute
+                  :z-index    5
+                  :elevation  5
+                  :max-height (/ max-y 2)})}
+        [mentions/autocomplete-mentions suggestions (:text-input-ref refs)]]))])
+
+(defn effect!
+  [{:keys [keyboard-shown] :as params}]
+  (rn/use-effect
+   (fn []
+     (when (or (not @keyboard-hiding?)
+               (and @keyboard-hiding? (not keyboard-shown)))
+       (reset! keyboard-hiding? false)
+       (if (not keyboard-shown)
+         (minimize params)
+         (update-y params))))))
+
+(defn prepare-params
+  [[refs window-height translate-y bg-opacity bg-bottom min-y max-y parent-height
+    max-parent-height chat-id suggestions reply edit images keyboard-shown]]
+  {:chat-id           chat-id
+   :translate-y       translate-y
+   :min-y             min-y
+   :max-y             max-y
+   :max-parent-height max-parent-height
+   :parent-height     parent-height
+   :suggestions       suggestions
+   :reply             reply
+   :edit              edit
+   :images            images
+   :refs              refs
+   :keyboard-shown    keyboard-shown
+   :bg-opacity        bg-opacity
+   :bg-bottom         bg-bottom
+   :window-height     window-height
+   :set-bg-opacity    (fn [value]
+                        (reanimated/set-shared-value bg-bottom (if (= value 1) 0 (- window-height)))
+                        (reanimated/set-shared-value bg-opacity (reanimated/with-timing value)))
+   :set-translate-y   (fn [value]
+                        (reanimated/set-shared-value translate-y (reanimated/with-timing value)))
+   :set-parent-height (fn [value]
+                        (reanimated/set-shared-value parent-height (reanimated/with-timing value)))})
+
+;; IMPORTANT! PLEASE READ BEFORE MAKING CHANGES
+
+;; 1) insets are not stable so we can't rely on the first values and we have to use it from render
```

Comment:
Yes, that's annoying. But can we just store them, once we get reliable value. I am doing something like that for shell height (as we can't get screen height reliably)
https://github.com/status-im/status-mobile/blob/220341e0be21765380270a579294162580facd10/src/status_im2/contexts/shell/view.cljs#L121
https://github.com/status-im/status-mobile/pull/14639#issuecomment-1366992260
https://github.com/status-im/status-mobile/pull/14639#issuecomment-1366995541
Perhaps we should calculate all screen parameters, (height, width, top, etc) correctly once, and just store them.

Reply:
```
[SafeAreaView](https://github.com/th3rdwave/react-native-safe-area-context#safeareaview) is the preferred way to consume insets. This is a regular View with insets applied as extra padding or margin. It offers better performance by applying insets natively and avoids flickers that can happen with the other JS based consumers

[useSafeAreaInsets](https://github.com/th3rdwave/react-native-safe-area-context#usesafeareainsets) offers more flexibility, but can cause some layout flicker in certain cases. Use this if you need more control over how insets are applied.
```

probably we could look at how SafeAreaView is implemented and just implement this in our native code